src/checker/Pulse.Typing.fst

```diff
@@ -978,6 +1006,21 @@ type st_typing : env -> st_term -> comp -> Type =
                                               condition_var = ppname_default } ))
                   (comp_while ppname_default inv)
 
+  | T_NuWhile:
+      g:env ->
+      inv:term ->
+      post_cond:term ->
+      cond:st_term ->
+      body:st_term ->
+      tot_typing g inv tm_slprop ->
+      tot_typing g (tm_exists_sl u0 (as_binder tm_bool) post_cond) tm_slprop ->
```

Comment:
This line seems a bit odd, should `post_cond` just have type `bool -> slprop`?

Reply:
This is a style we use elsewhere to close and type slprops that mention the return variable.
One could use any closing, e.g., lambda abstraction. tm_exists_sl works as well.